packages/beacon-dapp/src/dapp-client/DAppClient.ts

```diff
@@ -1549,8 +1560,8 @@ export class DAppClient extends Client {
       }
 
       return {
-        name: walletInfo.name,
-        icon: walletInfo.icon ?? selectedApp.logo,
+        name: selectedApp ? selectedApp.name : walletInfo.name,
```

Comment:
This means that we will use the "last selected" app over the name that we have stored in the wallet info. Is that what we want? We could do `name: walletInfo.name ?? selectedApp?.name`.